### PR TITLE
Display duration on start/edit view

### LINKED ISSUE #121: Properly display the current duration
## Description
We need to ensure that the duration on the top right corner of the card is correct
## Definition of done
- [ ] If the duration is set, we just use it along with the existing format method used in the log
- [ ] If there's no duration, we calculate the duration based on the current date and then use the existing format method
- [ ] If there's no duration, we need to update the number every second

### PR DESCRIPTION
**Pull request by Odrakir**
## Info
Displays the duration in the start edit view:
- When the TE is new and thus not started yet, it displays:  "00:00"
- When the TE is an existing stopped one it displays its duration in format: "h:mm:ss"
- When the TE is running (no duration) it displays "h:mm:ss" taking the current time from the time service

### Technical considerations
I changed the `formattedDuration` extension on `TimeInterval` which made no sense as it was taking a `Date` parameter.

### Relationships
Closes #121 

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

- [ ] The code used custom templates

### Pointers
Should be pretty simple to review

### Review template
- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones
- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Do it!

### CODE REVIEW

File: Modules/Timer/Timer/StartEdit/Views/StartEditViewController.swift [hunk 1]
```diff
@@ -128,6 +137,28 @@ public class StartEditViewController: UIViewController, Storyboarded, BottomShee
         }
 
         descriptionTextField.text = timeEntry.description
+        setDurationLabel(for: timeEntry)
+    }
+
+    private func setDurationLabel(for timeEntry: EditableTimeEntry) {
+
+        timer?.invalidate()
+        timer = nil
```
Comment (sofokli1) [inline Modules/Timer/Timer/StartEdit/Views/StartEditViewController.swift:146]: Should we maybe have this 2 lines in the view `didDisappear` as well?

File: Modules/Utils/Utils/Extensions/TimeInterval+Extensions.swift [hunk 2]
```diff
@@ -1,16 +1,12 @@
 import Foundation
 
 extension TimeInterval {
-    public func formattedDuration(ending endingDate: Date) -> String {
+    public func formattedDuration() -> String {
 
-        let startingDate = endingDate.addingTimeInterval(-self)
-        let calendar = Calendar.current
+        let formatter = DateComponentsFormatter()
```
Comment (sofokli1) [inline Modules/Utils/Utils/Extensions/TimeInterval+Extensions.swift:6]: can we extract this outside of the func. doesn't make sense to create the same thing every time we call the func.

File: Modules/Timer/Timer/StartEdit/Views/StartEditViewController.swift (hunk 1 above)
Comment (Odrakir) [reply Modules/Timer/Timer/StartEdit/Views/StartEditViewController.swift:146]: 🤔 I think I should just invalidate the timer on deinit, right? If I do it on didDisappear, when ever we send the app to background and come back the timer would be stopped

Comment (sofokli1) [reply Modules/Timer/Timer/StartEdit/Views/StartEditViewController.swift:145]: Even better

**Pull request merged**